notebooks/approx-nearest-neighbors.ipynb

```diff
@@ -440,6 +441,74 @@
     "# saves the LSH hashes for the input rows\n",
     "model.write().overwrite().save(\"/tmp/ann_model\")"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b1398af2",
+   "metadata": {},
+   "source": [
+    "## Spark Rapids ML (GPU CAGRA algorithm) \n",
+    "CAGRA is a cutting-edge graph-based algorithm available in cuVS, and is now integrated into the ApproximateNearestNeighbors class of Spark Rapids ML. Cagra currently supports sqeuclidean distance metric only, and the metric must be set before using the main APIs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0e0bef26",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "knn = ApproximateNearestNeighbors(k=2, algorithm='cagra', metric='sqeuclidean', algoParams={\"build_algo\" : \"nn_descent\"})\n",
```

Comment:
Perhaps in future pr we can consider (as in previous comment from @rishic3 ) how to make this more user friendly in terms of if cagra default to sqeuclidean, etc via some logic that is shared across init and setters.